mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1904,12 +1902,23 @@ where
             amount: big_decimal_from_sat_unsigned(kmd_rewards, coin.as_ref().decimals),
             claimed_by_me,
         };
-        (fee, Some(kmd_rewards_details))
+        (
+            big_decimal_from_sat(fee, coin.as_ref().decimals),
+            Some(kmd_rewards_details),
+        )
+    } else if input_amount == 0 {
+        let fee = verbose_tx.vin.iter().fold(0., |cur, input| {
+            let fee = match input {
+                TransactionInputEnum::Lelantus(lelantus) => lelantus.n_fees,
+                _ => 0.,
+            };
+            cur + fee
+        });
+        (fee.into(), None)
     } else {
         let fee = input_amount as i64 - output_amount as i64;
-        (fee, None)
+        (big_decimal_from_sat(fee, coin.as_ref().decimals), None)
     };
-    let fee = big_decimal_from_sat(fee, coin.as_ref().decimals);
```

Comment:
Both of these conditions can return just `fee: i64`.

Reply:
I've done it this way because `lelantus.n_fees` is f64 so it can be directly converted to BigDecimal. f64 -> i64 (sat_from_big_decimal) conversion can return an error, which handling I preferred to avoid.

Reply:
I see, thanks!